core/src/main/java/org/apache/calcite/sql/fun/SqlTrimFunction.java

```diff
@@ -130,6 +130,9 @@ public SqlTrimFunction(String name, SqlKind kind,
       if (operands[1] == null) {
         operands[1] = SqlLiteral.createCharString(" ", pos);
       }
+      if (operands[2] == null) {
+        throw new IllegalArgumentException("String to trim cannot be null");
```

Comment:
"The second argument of trim cannot be null"
It's not clear what "string" you are referring to.

Reply:
I think it's either the first or the third:

You can specify one argument (only the string to trim) or three (where to trim (leading, trailing both sides), what to trim from the string, the string to trim).

Reply:
> `"The second argument of trim cannot be null"`

This is not 100% correct I think as the string to trim from can be `null`: "trim('a' from cast(null as varchar(1)))" returns `null`, which is valid. I think it's more about the absence of the argument, which feels to me like a different thing than a value/field being `null`.

Ideally we should correctly fall through to the default case or handle the cases, where the string to trim is absent explicitly.